### DISCUSSION TITLE
docs: How do we conditionally populate relationships?

### DISCUSSION THREAD

**Discussion by tufailra97** [mikro-orm/mikro-orm#4511] [Q&A]
I am encountering an issue with Mikro ORM related to empty relationship conditions not returning any data.

I have three tables: "users," "accounts," and "user_profiles," with the following relationships:

```mermaid
erDiagram
    User ||--|{ UserProfile : More 
    Account ||--|{ UserProfile : More 
```

When I select all users and join their profiles without any sub condition, I get the expected data. However, if I add conditions to the relationships and there are no matching records, Mikro ORM does not return any data, leaving the result set empty/null.

I have defined the schema as follow:
```typescript
export class BaseEntity {
  @PrimaryKey()
  @Unique()
  id: string = v4();

  @Property()
  createdAt: Date = new Date();

  @Property({ onUpdate: () => new Date() })
  updatedAt: Date = new Date();
}

@Entity({ tableName: 'accounts' })
export class AccountEntity extends BaseEntity {
  @Property()
  name: string;

  @Property({ nullable: true })
  avatar?: string;

  @Property({ nullable: true, default: true, name: 'is_active' })
  isActive?: boolean;

  @OneToMany(() => UserProfileEntity, (userProfile) => userProfile.account)
  userProfiles = new Collection<UserProfileEntity>(this);
}

@Entity({ tableName: 'user_profiles' })
export class UserProfileEntity extends BaseEntity {
  @Property()
  username: string;

  @Property({ nullable: true })
  avatar?: string;

  @Property({ nullable: true, default: true, name: 'is_active' })
  isActive?: boolean;

  @ManyToOne()
  account: Rel<AccountEntity>;

  @ManyToOne()
  user: Rel<UserEntity>;

  @ManyToOne({ entity: () => RoleEntity })
  role: Rel<RoleEntity>;

  @Property({ nullable: true, name: 'is_verified' })
  isVerified?: boolean;
}


@Entity({ tableName: 'users' })
export class UserEntity extends BaseEntity {
  @Property({ name: 'first_name' })
  firstName: string;

  @Property({ name: 'last_name' })
  lastName: string;

  @Property()
  @Unique()
  email: string;

  @Property({ hidden: true })
  password: string;

  @OneToMany(() => UserProfileEntity, (userProfile) => userProfile.user)
  userProfiles = new Collection<UserProfileEntity>(this);
}
``` 

Here is an example of the query:

```typescript
const user = await this.em.findOne(
      UserEntity,
      {
        email: body.email,
        userProfiles: {
          isActive: true
        }
      },
      {
        populate: ['userProfiles']
      }
    );
```

Ideally, Mikro orm should return the user even if there a no user profile with `isActive` true.

**Comment by B4nan** [comment]
This is already documented here: 

https://mikro-orm.io/docs/nested-populate#filter-on-populated-entities
https://mikro-orm.io/docs/loading-strategies#population-where-condition

The default is to populate things regardless of the where condition (since v5), so it's weird that it does not behave that way for you. Can you provide a reproduction for that? What version are you on? What queries do you see when you enabled debug mode? Does specifying `populateWhere: {}` help?

I guess I should change the page url/title to something like "Populating relations", planning to do a bigger overhaul of the docs before v6 is out, so want to do things like that when it happens.

**Comment by tufailra97** [comment]
@B4nan Oh, my bad. I thought this was closed and not converted into a discussion. I ended up with raw queries and made quite a lot of progress. But let me see if I can go back through commits and reproduce the issue.
